drivers/hmis/app/graphql/mutations/ac_hmis/clear_mci.rb

```diff
@@ -24,13 +24,10 @@ def resolve(input:)
       return { errors: errors } if errors.any?
 
       mci = HmisExternalApis::AcHmis::Mci.new
-      response = mci.clearance(input.to_client)
+      response = mci.clearance(input.to_client, minimum_score: MATCH_THRESHOLD)
```

Comment:
Moved threshold filtering into MCI class, so we don't unnecessarily store mappings in the cache that we don't need. For example, clearance may return 50 results but only 5 of them are above the threshold to display to the user. This should also increase performance since we now only do ID/client lookups for the results that meet the threshold.